src/components/Toggle.astro

```diff
@@ -0,0 +1,125 @@
+---
+import type { Polymorphic } from 'astro/types'
+import Icon from 'fulldev-ui/components/Icon.astro'
+import Root from 'fulldev-ui/components/Root.astro'
+
+type Props<As extends 'div'> = Polymorphic<{ as: As }> & {
+  radius?: 'none' | 'auto' | 'full'
+  icon?: string
+  html?: string
+  text?: string
+  toggled: boolean
+}
+
+const {
+  as = 'div',
+  radius = 'auto',
+  name,
+  toggled,
+  icon,
+  html,
+  text,
+  ...rest
+} = Astro.props
+---
+
+<Root
+  class:list="toggle"
+  {as}
+  {radius}
+  {...rest}
+>
+  <label class="toggle-element">
```

Comment:
You have the Root and in that directly a label child. You do not need both, you can combine them into one.

set as="label" on Root and remove it from the props.

Reply:
Now I'm thinking about it, maybe you can even use the <Button> component, so you have the right styles right away. Set a as="label" on that.